### DISCUSSION TITLE
Issues using rft on Pyarmor 8.5.11

### DISCUSSION THREAD

**Discussion by Warfront1** [dashingsoft/pyarmor#1940] [Q&A] [answered]
I am in the process of updating from Pyarmor `8.5.6` to Pyarmor `8.5.11`.
I was able to get the build process working with the help of @jondy, as seen on the following discussion.
https://github.com/dashingsoft/pyarmor/discussions/1939

My code base has not changed since `8.5.6`, but I am now having problems running my application.
I believe I have isolated the problems to a regression with rft.

I am using the build process documented [above](https://github.com/dashingsoft/pyarmor/discussions/1939).
If I run the following Pyarmor command, the applications starts.
`pyarmor -d gen --pack test.spec --platform darwin.arm64,darwin.x86_64 -r ./src/`
Now if I use the following Pyarmor command, the application crashes.
`pyarmor -d gen --enable-rft --pack test.spec --platform darwin.arm64,darwin.x86_64 -r ./src/`

Here is a my python test script.
```Python
class MyService:
    def some_method(self):
        pass


class MyService2:
    def __init__(self, service):
        self.service = service

    def method1(self):
        print("method1")
        self.service.some_method()


class PyArmorTest:
    def __init__(self, my_interface):
        self.my_interface = my_interface

    def run(self):
        print("PyArmorTest")
        self.my_interface.method1()


print('test.py, this is __pyarmor__', __pyarmor__)
service = MyService()
interface = MyService2(service)
py_armor_test = PyArmorTest(interface)
py_armor_test.run()
```

Here is the error I am getting with rft enabled.
```
test.py, this is __pyarmor__ <built-in function __pyarmor__>
PyArmorTest
Traceback (most recent call last):
  File "<frozen __main__>", line 3, in <module>
  File "<frozen src.test>", line 28, in <module>
  File "<frozen src.test>", line 21, in pyarmor__7
AttributeError: 'pyarmor__3' object has no attribute 'method1'
[19109] Failed to execute script 'test' due to unhandled exception: 'pyarmor__3' object has no attribute 'method1'
[19109] Traceback:
Traceback (most recent call last):
  File "<frozen __main__>", line 3, in <module>
  File "<frozen src.test>", line 28, in <module>
  File "<frozen src.test>", line 21, in pyarmor__7
AttributeError: 'pyarmor__3' object has no attribute 'method1'
```

**Comment by jondy** [comment, marked as answer]
Please check this page to solve rft issue
https://pyarmor.readthedocs.io/en/latest/topic/rftmode.html

Also check this page for common solutions
https://pyarmor.readthedocs.io/en/latest/reference/solutions.html#target-device

**Reply by Warfront1** [reply]
Thanks, @jondy. I was able to get it working by simply excluding some methods.
It would be great if I didn’t have to exclude them, as they seemed to work with RFT in earlier versions of Pyarmor.
Fortunately, it only affects a small portion of my codebase, so overall, things are running well.
